### PR TITLE
readline(s), eachline: make `chomp=true` a keyword arg

### PR DESCRIPTION
**Pull request by StefanKarpinski**
This is a continuation of @mpastell's excellent work in https://github.com/JuliaLang/julia/pull/19944.

### CODE REVIEW

File: base/io.jl [hunk 1]
```diff
@@ -170,24 +170,48 @@ The text is assumed to be encoded in UTF-8.
...
+Read a single line of text from the given I/O stream or file (defaults to `STDIN`).
+When reading from a file, the text is assumed to be encoded in UTF-8. Lines in the
+input end with `'\\n'` or `"\\r\\n"` or the end of an input stream. When `chomp` is
+true (as it is by default), these trailing newline characters are removed from the
+line before it is returned. When `chomp` is false, they are returned as part of the
+line.
+"""
+function readline(filename::AbstractString; chomp::Bool=true)
+    open(filename) do f
+          readline(f, chomp=chomp)
```
Comment (tkelman) [inline base/io.jl]: nonstandard indent

**Comment by StefanKarpinski** [comment]
Tests passed, then I noticed a problem constructing `EachLine` objects. Fixed and added tests.

**Comment by nalimilan** [comment]
Thanks @mpastell for persisting through the multiple revisions!

**Comment by mpastell** [comment]
@nalimilan Thanks for the support and @StefanKarpinski thanks for finishing this! I have been extremely busy with my actual job the past week.

**Comment by StefanKarpinski** [comment]
> this reads a bit unclearly

I always take a pass through NEWS before a release to make sure everything is clear and included. Can fix this up then.

**Pull request merged**